Orm/Xtensive.Orm.PostgreSql/Orm.Providers.PostgreSql/SqlCompiler.cs

```diff
@@ -40,15 +41,15 @@ protected override SqlProvider VisitFreeText(FreeTextProvider provider)
       select.Columns.Add(SqlDml.Cast(fromTableRef.Columns[rankColumnName], SqlType.Double), rankColumnName);
       select.From = fromTableRef;
       if (provider.TopN == null) {
-        return CreateProvider(select, binding, provider);
+        return CreateProvider(select, new[] { binding }.Union(queryAndBindings.Bindings), provider);
```

Comment:
Probably yes

Reply:
By the way, ```queryAndBindings.Bindings``` can be not empty more frequently than you think, since every single entity hierarchy by default uses TypeId as constant value (which is now parameter).